test/Dialect/Calyx/errors.mlir

```diff
@@ -1050,3 +1050,101 @@ module attributes {calyx.entrypoint = "main"} {
     }
   }
 }
+
+// ----- 
+
+module attributes {calyx.entrypoint = "main"} {
+  calyx.component @main(%go: i1 {go}, %clk: i1 {clk}, %reset: i1 {reset}) -> (%done: i1 {done}) {
+    %r.in, %r.write_en, %r.clk, %r.reset, %r.out, %r.done = calyx.register @r : i32, i1, i1, i1, i32, i1
+    calyx.wires {
+
+    }
+    calyx.control {
+      // expected-error @+1 {{'calyx.invoke' op should specify how one of input or output ports for data transfer is to be connected, which belong to instance 'r'.}}
+      calyx.invoke@r() -> ()
+    }
+  }
+}
+
+// ----- 
+
+module attributes {calyx.entrypoint = "main"} {
+  calyx.component @main(%go: i1 {go}, %clk: i1 {clk}, %reset: i1 {reset}) -> (%done: i1 {done}) {
+    %c10 = hw.constant 10 : i32
+    %r.in, %r.write_en, %r.clk, %r.reset, %r.out, %r.done = calyx.register @r : i32, i1, i1, i1, i32, i1 
+    calyx.wires {
+
+    }
+    calyx.control {
+      // expected-error @+2 {{'calyx.invoke' op has a cell port as the destination with the incorrect direction.}} 
```

Comment:
Where is this error message being produced?

Reply:
You can see [here](https://github.com/llvm/circt/blob/72672984a7fa4092d2295a010527d647c2893ae3/lib/Dialect/Calyx/CalyxOps.cpp#L1421).

Reply:
Ah thank you, so this existed before your work.